### DISCUSSION TITLE
DataFlow - access to directus_* tables

### DISCUSSION THREAD

**Discussion by mslonina** [directus/directus#13716] [Feature Requests]
Hi,

thanks again for the long awaited DataFlows! I noticed however, that for the operation create/update/delete data only the custom models are available. I could imagine, for example the need to update `directus_users` (like setting value of some custom field) or maybe changing permissions - is it by design not having access to `directus_*` tables within the operations?

**Comment by rijkvanzanten** [comment]
The current item-* operations rely on the ItemsService exclusively, which means that certain custom logic pieces for the system tables won't fire:

https://github.com/directus/directus/blob/6e87d8a55b67a2994297014dc9da125e7a4d1bdb/api/src/operations/item-create/index.ts#L33-L37

In order to safely allow CRUD on the system tables as well, we should add a similar service-selector as we're using in GraphQL:

https://github.com/directus/directus/blob/6e87d8a55b67a2994297014dc9da125e7a4d1bdb/api/src/services/graphql.ts#L1612-L1651

**Comment by mslonina** [comment]
I've tried to workaround this by using a webhook, yet it failed, see #13737.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
